iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/functionclass/EventDataListener.java

```diff
@@ -154,171 +148,241 @@ public EventDataListener(final Device device, Scheduler scheduler, DatabaseUtils
+                            this.databaseUtils.updateDatabaseSchedule(
+                                    this.device.getRelay(relayIndex).getSchedule(scheduleIndex));
+
+                            modified = true;
+                        } catch (final Exception e) {
+                            logger.warn("Schedule exception: ", e);
+                        }
                     }
                     break;
                 }
 
                 case "tOnT": {
-                    logger.info("Time On Type found.");
                     final int relayIndex = this.extractRelayIndex(bda.getReference());
                     final int scheduleIndex = this.extractScheduleIndex(bda.getReference());
-                    try {
-                        this.device.getRelay(relayIndex).getSchedule(scheduleIndex)
-                                .setTimeOnTypeInt(((BdaInt8) bda).getValue());
-                        modified = true;
-                    } catch (final Exception e) {
-                        logger.info("Schedules above 50 are not implemented in the GXF platform. Skip ...");
+
+                    if (scheduleIndex <= 50) {
+                        logger.info("Time On Type found.");
+                        try {
+                            this.device.getRelay(relayIndex).getSchedule(scheduleIndex)
+                                    .setTimeOnTypeInt(((BdaInt8) bda).getValue());
+
+                            // Sends updated Schedule to database
+                            this.databaseUtils.updateDatabaseSchedule(
+                                    this.device.getRelay(relayIndex).getSchedule(scheduleIndex));
+
+                            modified = true;
+                        } catch (final Exception e) {
+                            logger.warn("Schedule exception: ", e);
+                        }
                     }
                     break;
                 }
 
                 case "tOff": {
-                    logger.info("Time Off found.");
                     final int relayIndex = this.extractRelayIndex(bda.getReference());
                     final int scheduleIndex = this.extractScheduleIndex(bda.getReference());
-                    try {
-                        this.device.getRelay(relayIndex).getSchedule(scheduleIndex);
-                        int timeInt = (((BdaInt32) bda).getValue());
-                        int timeHour = timeInt / 100;
-                        int timeMinute = timeInt % 100;
-                        LocalTime timeLocalTime = LocalTime.of(timeHour, timeMinute);
-                        this.device.getRelay(relayIndex).getSchedule(scheduleIndex).setTimeOff(timeLocalTime);
-                        modified = true;
-                    } catch (final Exception e) {
-                        logger.info("Schedules above 50 are not implemented in the GXF platform. Skip ...");
+
+                    if (scheduleIndex <= 50) {
```

Comment:
This check is copied a number of times. Other parts of the case block also look a lot alike. Check how you can eliminate all this duplication.

Reply:
Done.